vmdk_plugin/main.go

```diff
@@ -91,40 +91,62 @@ func logInit(logLevel *string, logFile *string, configFile *string) bool {
 // main for docker-volume-vsphere
 // Parses flags, initializes and mounts refcounters and finally services Docker requests
 func main() {
-	// connect to this socket
-	port := flag.Int("port", 1019, "Default port for vmci")
-	useMockEsx := flag.Bool("mock_esx", false, "Mock the ESX server")
+	var driver volume.Driver
+
+	// Define command line options
+	driverName := flag.String("driver", "vmdk", "Volume driver")
 	logLevel := flag.String("log_level", "info", "Logging Level")
 	configFile := flag.String("config", config.DefaultConfigPath, "Configuration file path")
-	flag.Parse()
 
-	vmdkops.EsxPort = *port
+	// photon driver options
+	targetURL := flag.String("target", "", "Photon controller URL")
+	projectID := flag.String("project", "", "Project ID of the docker host")
+	vmID := flag.String("host", "", "ID of docker host")
+
+	// vmdk driver options
+	port := flag.Int("port", 1019, "Default port to connect to ESX service")
+	useMockEsx := flag.Bool("mock_esx", false, "Mock the ESX service")
+
+	flag.Parse()
 
 	logInit(logLevel, nil, configFile)
 
 	log.WithFields(log.Fields{
-		"version":   version,
-		"port":      vmdkops.EsxPort,
-		"mock_esx":  *useMockEsx,
+		"driver":    *driverName,
 		"log_level": *logLevel,
 		"config":    *configFile,
-	}).Info("Docker VMDK plugin started ")
+	}).Info("Starting plugin ")
+
+	log.WithFields(log.Fields{
+		"target":  *targetURL,
+		"project": *projectID,
+		"host":    *vmID,
+		"port":    *port}).Info("Plugin options - ")
+
+	if *driverName == "photon" {
+		driver = photon.NewVolumeDriver(*targetURL, *projectID,
```

Comment:
This topic is still not closed and hence the way its been coded. For now if any of these params are not given the client simply fails to connect to the photon controller. Plan is that when PC deploys a VM it will create the config file for the plugin to use.